### PR TITLE
chore: new AST based script supporting current markdown for LLMs

### PR DESCRIPTION
**Pull request by jeremenichelli**
In this PR:
 - We move the generation script for simplified markdown to an AST approach.
 - Additionally we add support for a specialized schema resolution on React components.
 - We add a first group of components to bring the files on parity with current offering.

## Summary by CodeRabbit

* **Refactor**
  * Reworked guide markdown generation to use structured parsing, inline referenced partials, and improve internal link handling and header formatting.
  * Added more robust error handling with graceful fallback when generation encounters issues.
* **New Features**
  * Added rendering support for **Admonition**, **Link**, **Panel**, **StepHike**, and **TabPanel** components in guide content.

### CODE REVIEW

**Review by daltino — COMMENTED** [review]
Great direction! Moving to an AST-based approach is significantly more robust than string manipulation, and the component schema pattern is clean and extensible. I have a few observations and suggestions before merge.

---

### `generate-guides-markdown.ts`

**Positive:** Switching to `fs/promises` is a nice modernization that fits well with the broader AST refactor.

#### Suggestion — Error handling for async file reads

With the move to promise-based filesystem APIs, make sure file reads are handled appropriately. In a glob-driven content pipeline, a single malformed or unreadable file ideally shouldn't terminate the entire generation process unless that behavior is intentional.

For example, consider wrapping per-file processing in a `try/catch` and reporting failures with enough context to identify the problematic file.

---

### `markdown-schema/Admonition.ts`

```ts
const capitalize = <T extends string>(s: T) =>
  (s[0].toUpperCase() + s.slice(1)) as Capitalize<typeof s>
```

#### Edge case — Empty string

If `props.type` can ever be an empty string, `s[0]` will be `undefined`, producing unexpected output.

A defensive implementation could be:

```ts
const capitalize = <T extends string>(s: T) =>
  s.length === 0
    ? s
    : ((s[0].toUpperCase() + s.slice(1)) as Capitalize<typeof s>)
```

This may not be necessary if upstream validation guarantees a non-empty value, but it's a cheap safeguard.

#### Minor — Intentional flattening

The output format appears to reduce admonitions to something like:

```md
Note: content
```

rather than preserving block-level structure.

For LLM-oriented output, that simplification may be entirely intentional. If so, a short comment explaining the rationale would help future maintainers avoid "improving" it back into a more complex representation.

---

### `markdown-schema/Link.ts`

#### Edge case — Empty link text

`props.href ?? ''` is already handled safely, and the URL normalization path looks fine.

One small consideration: if `children` is empty, the serializer would emit:

```md
[](url)
```

which is valid Markdown but not especially useful for LLM consumption.

A possible alternative:

```ts
return children ? `[${children}](${href})` : href
```

This preserves the destination while avoiding empty link text.

---

### `markdown-schema/Panel.ts`

```ts
export const Panel = ({ props }: { props: Record<string, unknown> }): string => {
  return String(props.title)
}
```

#### Edge case — Missing title

If `title` is absent, the current implementation emits the literal string `"undefined"`.

Consider:

```ts
export const Panel = ({ props }: { props: Record<string, unknown> }): string => {
  return props.title != null ? String(props.title) : ''
}
```

#### Question — Children are discarded

The component currently ignores `children` entirely.

If panels are being intentionally collapsed into headings or summaries for token efficiency, that makes sense. However, if panel bodies contain meaningful documentation content, this could result in information loss in the generated markdown.

A brief comment documenting the intended behavior would be helpful either way.

---

### `markdown-schema/StepHike.ts`

The overall structure is clear and easy to follow.

#### `Step` — Assumption about `props.step`

The implementation appears to assume `props.step` is a numeric value used to generate output such as:

```md
1. Step content
```

If non-numeric values are possible, output like:

```md
intro. Step content
```

may be surprising.

If the source schema guarantees numeric values, no change is needed. Otherwise, consider validating or normalizing the value before rendering.

#### `Details` — Newline normalization

The pattern:

```ts
`**${title}**\n\n${children}`
```

assumes `children` does not already contain leading whitespace or blank lines.

This is probably low risk, but it may be worth normalizing child content if inconsistent spacing becomes an issue in generated output.

---

### Testing

No tests are visible in the PR.

Given that this code sits in a content-generation pipeline feeding LLMs, I'd strongly recommend adding at least a small amount of coverage:

#### Component-level tests

Validate behavior for:

* `Admonition`
* `Link`
* `Panel`
* `Step`
* `Details`

including the edge cases discussed above.

#### Integration / Snapshot tests

A fixture-based test would provide particularly strong protection against regressions:

```text
MDX input
   ↓
AST
   ↓
Simplified markdown output
```

A handful of representative fixtures and snapshots would make future refactors much safer.

---

### Summary

The overall architecture is solid. Moving from string-based transformations to an AST-driven pipeline should improve maintainability and correctness, and the component handler pattern is straightforward to extend.

The main items I'd like to clarify before approving are:

1. Whether `Panel` intentionally drops child content or if this is an accidental loss of context.
2. Whether missing values can result in literal `"undefined"` appearing in generated output.
3. Whether component edge cases (empty strings, missing props, empty link text) are guaranteed impossible by upstream validation.
4. Whether there is test coverage planned for the markdown generation pipeline, particularly integration-level fixtures.

Assuming the content-loss question around `Panel` is addressed and some test coverage is added, this looks like a strong improvement over the previous implementation.

**Review by ChrisChinchilla — APPROVED** [review]

File: apps/docs/internals/generate-guides-markdown.ts [hunk 1]
```diff
@@ -1,198 +1,160 @@
...
-      .slice(i + 1, closeIdx)
-      .replace(/<[^>]+>/g, ' ')
-      .replace(/\s+/g, ' ')
-      .trim()
+  const headerParts: string[] = []
+  if (data.title) headerParts.push(`# ${data.title}`)
+  if (data.subtitle) headerParts.push(String(data.subtitle))
+  // Add description only when differs from subtitle.
+  if (data.description && String(data.description) !== String(data.subtitle))
+    headerParts.push(String(data.description))
```
Comment (gregnr) [inline apps/docs/internals/generate-guides-markdown.ts:153]: Since we're already serializing markdown AST via `serializeMdx()`, perhaps it would be simpler/cleaner to add these titles/headings as actual AST `heading` nodes prior to serialization vs after?

File: apps/docs/internals/markdown-schema/Admonition.ts
Comment (gregnr) [inline apps/docs/internals/markdown-schema/Admonition.ts:1]: Rather than creating a separate directory (`markdown-schema`) for all of these components, what if we created a `Admonition.static.ts` or `Admonition.md.ts` that lives next to the real component? Co-locating these in the same place will likely make it easier to maintain.

File: apps/docs/internals/generate-guides-markdown.ts (hunk 1 above)
Comment (jeremenichelli) [reply apps/docs/internals/generate-guides-markdown.ts:153]: Hmm, that's interesting to see if it actually brings benefits, I would say no because if after this the tree gets traversed  then you are just joining two strings which is more performant that making the tree larger before evaluation. 

I will run some tests tomorrow and see if there's a difference.

File: apps/docs/internals/markdown-schema/Admonition.ts
Comment (jeremenichelli) [reply apps/docs/internals/markdown-schema/Admonition.ts:1]: I actually thought of that, and in the future we might do that, I am trying to encapsulate functionality so it's easier to port in the future if we do a re-write. Files within the project are all over the place, client and server components are mixed, some components carry data, some do data fetching. So it's a bit chaotic in my opinion. And spread all files with one single purpose across different directories might make that worse.

I think I would consider that once we have a cleaner and more predictable folder structure within the project so the overhead is not big for the import mechanism, and for humans to scan these components.

File: apps/docs/internals/generate-guides-markdown.ts (hunk 1 above)
Comment (gregnr) [reply apps/docs/internals/generate-guides-markdown.ts:153]: Yeah the concern was less about performance and more about maintenance / reducing foot guns (since we're already working in AST, we can take advantage of its benefits - type safety, etc)
